server/src/main/java/dev/slimevr/tracking/trackers/Tracker.kt

```diff
@@ -84,6 +121,53 @@ class Tracker @JvmOverloads constructor(
 // 		}
 	}
 
+	/**
+	 * If 0 then it's null
+	 */
+	var lastResetStatus = 0u
+	private fun reportRequireReset() {
+		require(lastResetStatus == 0u) {
+			"lastResetStatus must be 0u, but was $lastResetStatus"
+		}
+
+		val tempTrackerNum = this.trackerNum
+		val statusMsg = StatusTrackerResetT().apply {
+			trackerId = TrackerIdT().apply {
+				if (device != null) {
+					deviceId = DeviceIdT().apply { id = device.id }
+				}
+				trackerNum = tempTrackerNum
+			}
+		}
+		val status = StatusDataUnion().apply {
+			type = StatusData.StatusTrackerReset
+			value = statusMsg
+		}
+		lastResetStatus = vrServer.statusSystem.addStatus(status, true)
+	}
+
+	var lastErrorStatus = 0u
+	private fun reportErrorStatus() {
+		require(lastErrorStatus == 0u) {
+			"lastResetStatus must be 0u, but was $lastErrorStatus"
+		}
+
+		val tempTrackerNum = this.trackerNum
+		val statusMsg = StatusTrackerErrorT().apply {
+			trackerId = TrackerIdT().apply {
+				if (device != null) {
+					deviceId = DeviceIdT().apply { id = device.id }
+				}
+				trackerNum = tempTrackerNum
+			}
+		}
+		val status = StatusDataUnion().apply {
+			type = StatusData.StatusTrackerError
+			value = statusMsg
+		}
+		lastErrorStatus = vrServer.statusSystem.addStatus(status, true)
+	}
```

Comment:
This feels like a little awkward of a place to have these?

Reply:
the problem is the variable used to hold the latest status reported by the own tracker